lib/iris/cube.py

```diff
@@ -2302,7 +2306,8 @@ def _as_list_of_coords(self, names_or_coords):
         Convert a name, coord, or list of names/coords to a list of coords.
         """
         # If not iterable, convert to list of a single item
-        if not hasattr(names_or_coords, '__iter__'):
+        if (not hasattr(names_or_coords, '__iter__') or
+                isinstance(names_or_coords, str)):
```

Comment:
Could these checks be simplified to just checking if `names_or_coords` is a `str`? This will be true regardless of Python 2 or 3...